src/smallmap_gui.cpp

```diff
@@ -1004,6 +1005,43 @@ class SmallMapWindow : public Window {
 		}
 	}
 
+	/**
+	 * Adds industry names to the smallmap.
+	 * @param dpi the part of the smallmap to be drawn into
+	 */
+	void DrawIndustryNames(const DrawPixelInfo *dpi) const
+	{
+		if (this->map_type != SMT_INDUSTRY) return;
+
+		for (const Industry *i : Industry::Iterate()) {
+			const LegendAndColour &tbl = _legend_from_industries[_industry_to_list_pos[i->type]];
+			if (!tbl.show_on_map) continue;
+
+			const bool is_blinking = i->type == _smallmap_industry_highlight && !_smallmap_industry_highlight_state;
+			if (is_blinking) continue;
+
+			const uint8_t legend_text_width = GetStringBoundingBox(tbl.legend, FS_SMALL).width;
```

Comment:
Should I cache this somewhere? e.g. `_industry_to_legend_text_width` similar to `_industry_to_list_pos`:

https://github.com/OpenTTD/OpenTTD/blob/e35680d0cbf8fe8e7c4b55729a45e9b0257f09fa/src/smallmap_gui.cpp#L173-L174